src/components/Root/LegacyRouteRedirection.tsx

```diff
@@ -10,14 +10,28 @@ type Props = {
   history: History
 }
 
+enum SHORT_NAME {
+  GNOSIS_CHAIN = 'gno',
+  XDAI = 'xdai',
+}
+
 const LEGACY_SAFE_ADDRESS_SLUG = 'safeAddress'
 
 const LegacyRouteRedirection = ({ history }: Props): ReactElement | null => {
   const { pathname, hash, search } = window.location
 
   const isLegacyRoute = pathname === `${PUBLIC_URL}/` && hash.startsWith('#/')
 
-  if (!isLegacyRoute) return null
+  if (!isLegacyRoute) {
+    const XDAI_ROUTE_PREFIX = `${PUBLIC_URL}/${SHORT_NAME.XDAI}:`
+    const GNO_ROUTE_PREFIX = `${PUBLIC_URL}/${SHORT_NAME.GNOSIS_CHAIN}:`
+
+    const hasxDaiShortName = pathname.includes(XDAI_ROUTE_PREFIX)
+    if (hasxDaiShortName) {
+      history.replace(`${pathname.replace(XDAI_ROUTE_PREFIX, GNO_ROUTE_PREFIX)}${search}`)
+    }
+    return null
```

Comment:
If it's not targeting the old hash URLs, why is this redirect in `LegacyRouteRedirection.tsx`?
It could just be a regular redirect route.

Reply:
I still don't get why it must be so complicated and not just a regular redirect route.
This works just fine:

```
      <Route
        // Redirect xdai to gno
        path="/xdai\::url*"
        render={() => <Redirect to={`/app` + location.pathname.replace(/\/xdai:/, '/gno:')} />}
      />
```